src/cryptography/hazmat/backends/openssl/rsa.py

```diff
@@ -523,6 +534,15 @@ def public_bytes(self, encoding, format):
         )
 
     def verify(self, signature, data, padding, algorithm):
-        verifier = self.verifier(signature, padding, algorithm)
-        verifier.update(data)
-        verifier.verify()
+        padding_enum = _rsa_sig_determine_padding(
+            self._backend, self, padding, algorithm
+        )
+        data, algorithm = _calculate_digest_and_algorithm(
+            self._backend, data, algorithm
+        )
+
+        _check_hashed_data_length(data, algorithm)
```

Comment:
I think it'd make sense to just fold `_check_hashed_data_length` into `_calculate_digest_and_algorithm`